### ISSUE TITLE
Code formatting issues with square brackets

### ISSUE THREAD

**Issue by ajmas** [MacDownApp/macdown#892]
I have the following Markdown code:

  ```
  {
    "name": "",
    "description": ""
    "entries": [{
      
    }]
  ]
  ```
This formats correctly in GitHub issue tracker, but not in Macdown. The attached image illustrates the behaviour

![screen shot 2017-12-04 at 17 17 59](https://user-images.githubusercontent.com/682269/33579249-21d4a21a-d917-11e7-92b2-38ea5e0a9850.png)

MacDown 0.7.1, macOS 10.13.1

**Comment by franklinyu** [comment]
Cannot reproduce with either fenced block or indented block.

**Comment by ajmas** [comment]
Turns out It was an incomplete scenario. It happens when it is a child of a bullet list:

![screen shot 2017-12-06 at 11 22 29](https://user-images.githubusercontent.com/682269/33672201-a9e619e8-da77-11e7-9a3c-f2a1050c240a.png)

Also an issue in master branch version.

**Comment by franklinyu** [comment]
That's a great example. It seems like Hoedown assumes that indentation level is 4; in your example, if the code block is indented with four spaces, it will show just fine.

However, we are planning to move to CommonMark, where it is [clearly defined](http://spec.commonmark.org/0.28/#example-219) that this is not desired, and your case should be correctly recognized as indentation with two spaces. In conclusion, this is a bug, but need some time to fix.

**Comment by franklinyu** [comment]
@cori Does the fenced code belong to the item or not?

**Comment by cori** [comment]
Ideally (for me in this case) it would be a part of that bullet, but would display formatted

**Comment by franklinyu** [comment]
Then you should indent the fenced code block, or the parser would got confused.

**Comment by cori** [comment]
same result when indented 4 spaces:
![image](https://user-images.githubusercontent.com/46317/35952650-a776c382-0c46-11e8-8c6b-da631c5fb622.png)

**Comment by cori** [comment]
here's the actual content
```
{
	"prop1": "thing1",
	"prop2": "thing2",
	"obj_prop": {
		"oprop1": "othing1",
		"oprop2": "othing2"
	}
}
```

* bullet item
    ```
	{
		"prop1": "thing1",
		"prop2": "thing2",
		"obj_prop": {
			"oprop1": "othing1",
			"oprop2": "othing2"
		}
	}
	```

**Comment by franklinyu** [comment]
You need a blank line between the code block and everything else.

**Comment by franklinyu** [comment]
Oops, looks like CommonMark allows this:

> Fenced code blocks can interrupt paragraphs, and can be followed directly by paragraphs, without a blank line between…

But that’s another bug.

**Comment by uranusjr** [comment]
Yeah CommonMark is more lenient with empty lines. The produced HTML is the same, however, since `<pre>` is a block-level element, so I would recommend users to add an empty line anyway. There’s not such thing as a code block belonging to a paragraph.

**Comment by franklinyu** [comment]
The `develop` branch of Hoedown has it right again. So we are back to the old issue #704. That one is blocking. I think it’s better to write well-formed Markdown instead of trying the edge-cases, until we find a better Markdown implementation.
